feat(curriculum): add Sudoku solver project Python (#52452)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-classes-and-objects-by-building-a-sudoku-solver/6569fd01dab2ea547d98f093.md

---
id: 6569fd01dab2ea547d98f093
title: Step 75
challengeType: 20
dashedName: step-75
---

# --description--

> Use the solver method with the Board object to check if it returns `True`.

Reviewer: ```suggestion
Create an `if` statement that checks if the `solver()` method call from the `gameboard` object returns `True`.

Then, add a `print()` call inside the `if` block passing the following string: `'\nSolved puzzle:'`.
```

# --hints--

> You should use an `if` statement to check if the solver method returns `True`.

Reviewer: ```suggestion
You should create an `if` statement that checks if the `solver()` method call from the `gameboard` object returns `True`.
```

```js

> assert.match(code, /if/);

Reviewer: ```suggestion
({ test: () => assert.match(e.code.original["main.py"], /^\s{4}if\s+gameboard\.solver\s*\(\s*\)\s*:/ms) })
```
Changed this and the following test to include the print call in this step.

```

> Use the solver method(`solver()`) with the Board object(`gameboard`) to check if it returns `True`.

Reviewer: ```suggestion
You should call `print` passing the provided string inside the new `if` block.
```

```js

> assert.match(code, /if\s+gameboard\.solver\(\)/);

Reviewer: ```suggestion
({ test: () => assert.match(e.code.original["main.py"], /^(\s{4})if\s+gameboard\.solver\s*\(\s*\)\s*:\s*^\1\1print\s*\(\s*("|')\\nSolved\spuzzle:\2\s*\)/m) })
```
